### ISSUE TITLE
Nargo prove cannot be run in parallel

### ISSUE THREAD

**Issue by 0x73696d616f** [noir-lang/noir#4624]
### Problem

I want to run tests in parallel calling nargo prove but can't do it easily due to having to write to Prover.toml and read from the .proof file.

### Happy Case

I forked the repo a built a working solution here https://github.com/0x73696d616f/noir.

### Project Impact

Nice-to-have

### Workaround

None

### Would you like to submit a PR for this Issue?

None

**Comment by 0x73696d616f** [comment]
it seems that there is some lock somewhere that prevents prove from being in executed in parallel.
Blocking waiting for file lock on build directory\n

**Comment by jfecher** [comment]
The name of this PR sounds unrelated to the parallelism issue

**Comment by 0x73696d616f** [comment]
Yes, thank you.
Originally I thought I would be able to run nargo prove in parallel by adding --input and --show-output flags so I did not have concurrency issues when reading the same file from nargo prove commands in parallel.
However, it seems that nargo prove can't be run in parallel anyway
